src/mlpack/tests/cmaes_test.cpp

```diff
@@ -0,0 +1,149 @@
+using namespace mlpack::regression;
+
+BOOST_AUTO_TEST_SUITE(CMAESTest);
+
+// a simple rosenbrock function implemented
+class rosenbrockFunc
+{
+ public:
+  int N;
+
+  rosenbrockFunc(int x){ N = x-1; }
+
+  double NumFunctions(){return N;}
+
+  double Evaluate(arma::mat& x, int i)
+  {
+    return 100.*pow((pow((x[i]), 2) - x[i+1]), 2)
+    + pow((1.-x[i]), 2);
+  }
+};
+
+
+BOOST_AUTO_TEST_CASE(SimpleCMAESTestFunction)
+{
+  SGDTestFunction test;
+
+  size_t N = test.NumFunctions();
+
+  CMAES s(N, 0.5, 0.3, 10000, 1e-13, 1e-13);
+
+  arma::mat coordinates(N, 1);
+  double result = s.Optimize(test, coordinates);
+
+  BOOST_REQUIRE_CLOSE(result, -1.0, 0.05);
+  BOOST_REQUIRE_SMALL(coordinates[0], 1e-3);
+  BOOST_REQUIRE_SMALL(coordinates[1], 1e-7);
+  BOOST_REQUIRE_SMALL(coordinates[2], 1e-7);
+}
+
+BOOST_AUTO_TEST_CASE(LogisticRegressionTestWithCMAES)
+{
+  // Generate a two-Gaussian dataset.
+  GaussianDistribution g1(arma::vec("1.0 1.0 1.0"), arma::eye<arma::mat>(3, 3));
+  GaussianDistribution g2(arma::vec("9.0 9.0 9.0"), arma::eye<arma::mat>(3, 3));
+
+  arma::mat data(3, 1000);
+  arma::Row<size_t> responses(1000);
+  for (size_t i = 0; i < 500; ++i)
+  {
+    data.col(i) = g1.Random();
+    responses[i] = 0;
+  }
+  for (size_t i = 500; i < 1000; ++i)
+  {
+    data.col(i) = g2.Random();
+    responses[i] = 1;
+  }
+
+  // Shuffle the dataset.
+  arma::uvec indices = arma::shuffle(arma::linspace<arma::uvec>(0,
+      data.n_cols - 1, data.n_cols));
+  arma::mat shuffledData(3, 1000);
+  arma::Row<size_t> shuffledResponses(1000);
+  for (size_t i = 0; i < data.n_cols; ++i)
+  {
+    shuffledData.col(i) = data.col(indices[i]);
+    shuffledResponses[i] = responses[indices[i]];
+  }
+
+  // Create a test set.
+  arma::mat testData(3, 1000);
+  arma::Row<size_t> testResponses(1000);
+  for (size_t i = 0; i < 500; ++i)
+  {
+    testData.col(i) = g1.Random();
+    testResponses[i] = 0;
+  }
+  for (size_t i = 500; i < 1000; ++i)
+  {
+    testData.col(i) = g2.Random();
+    testResponses[i] = 1;
+  }
+
+  int dim = shuffledData.n_rows + 1;
+  arma::mat start1(dim, 1); start1.fill(0.5);
+  arma::mat initialStdDeviations1(dim, 1); initialStdDeviations1.fill(1.5);
+
+  CMAES test1(dim, 05, 0.3, 10000, 1e-10, 1e-10);
+
+  LogisticRegression<> lr(shuffledData, shuffledResponses, test1, 0.5);
+
+  // Ensure that the error is close to zero.
+  const double acc = lr.ComputeAccuracy(data, responses);
+  BOOST_REQUIRE_CLOSE(acc, 100.0, 0.3); // 0.3% error tolerance.
+
+  const double testAcc = lr.ComputeAccuracy(testData, testResponses);
+  BOOST_REQUIRE_CLOSE(testAcc, 100.0, 0.6); // 0.6% error tolerance.
+}
+
+BOOST_AUTO_TEST_CASE(rosenbrockFunctionCMAES)
+{
+  for (int i = 5; i < 50; i+=5)
+  {
+    rosenbrockFunc test(i);
+
+    CMAES s(i, 0.5, 0.3, 100000, 1e-16, 1e-15);
+
+    arma::vec coordinates(i);
+    double result = s.Optimize(test, coordinates);
+
+    for (int j = 0; j < i-1; j++)
+      BOOST_REQUIRE_CLOSE(coordinates[j], 1, 10);
```

Comment:
The last parameter of ``BOOST_REQUIRE_CLOSE``specifies the absolute value of difference between the first two parameter, which should be around ``0.01``, a value of 10 would mean that ``coordinates[0] = 5`` would also pass the check.